tool/python/README.md

```diff
@@ -0,0 +1,305 @@
+    SINGAROOT/tool/python
+    |-- pb2 (has job_pb2.py)
+    |-- singa 
+        |-- model.py 
+        |-- layer.py 
+        |-- parameter.py 
+        |-- utils 
+            |-- utility.py 
+            |-- message.py 
+        |-- datasets 
+            |-- cifar10.py 
+            |-- mnist.py 
+            |-- rbm.py 
+            |-- ae.py 
+    |-- examples 
+        |-- cifar10_cnn.py, mnist_mlp.py, , mnist_rbm1.py, rnnlm_usermodel.py, etc. 
+
+
+### How to Run
+```
+bin/singa-run.sh -exe user_main.py -conf cluster.conf
+```
+The python code, i.e., user_main.py, would create the JobProto object and pass it to Driver::Train.
+But ./bin/singa-run.sh needs to get the cluster topology, hence we still need to pass a cluster.conf to it.
+The cluster.conf has the configuration for a JobProto with all other fields empty except the cluster field.
+
+
+#### Examples
+```
+cd SINGA_ROOT
+bin/singa-run.sh -exe tool/python/examples/cifar10_cnn.py -conf tool/python/examples/cluster.conf
+```
+
+### Layer class (inherited)
+
+* Data
+* Dense
+* Activation
+* Convolution2D
+* MaxPooling2D
+* AvgPooling2D
+* LRN2D 
+* Dropout
+* Encoder
+* Decoder
+* RBM
+
+#### for user defined layers 
+
+The following classes are designed to include user-defined layers for RNNLM example.
+
+* Embedding
+* RNNLM
+* UserLossRNNLM
+
+
+### Parameter class
+
+* Parameter 
+
+### Model class
+
+* Model class has `jobconf` (JobProto) and `layers` (layer list)
+
+Methods in Model class
+
+* add
+	* add Layer into Model
+
+* compile	
+	* set Updater (i.e., optimizer) and Cluster (i.e., topology) components
```

Comment:
currently, we use 'cluster.conf' for cluster topology, but I still expect users to set 'workspace'.
So, I will comment it out now, and modify it. OK?